src/middlewares/auth.middleware.js

```diff
@@ -0,0 +1,30 @@
+import { ApiError } from '../exeptions/api.error.js';
+import { jwtService } from '../services/index.js';
+
+export const authMiddleware = (req, res, next) => {
+  const authorization = req.headers['authorization'];
+  const throwUnauthorizedError = () => {
+    throw ApiError.unauthorized({
+      user: 'unauthorized user',
+    });
+  };
+
+  if (!authorization) {
+    throwUnauthorizedError();
+  }
+
+  const [, token] = authorization?.split(' ');
+
+  if (!token) {
+    throwUnauthorizedError();
+  }
+
+  const userData = jwtService.verify(token);
```

Comment:
Ensure that `jwtService.verify` is correctly implemented to handle token verification. If the token is invalid, it should return `null` or throw an error, which is handled by `throwUnauthorizedError`.